flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorContext.java

```diff
@@ -257,9 +257,23 @@ public void signalNoMoreSplits(int subtask) {
         notifier.notifyReadyAsync(callable, handler);
     }
 
+    /** {@inheritDoc} If the runnable throws an Exception, the corresponding job is failed. */
     @Override
     public void runInCoordinatorThread(Runnable runnable) {
-        coordinatorExecutor.execute(runnable);
+        coordinatorExecutor.execute(wrap(runnable));
```

Comment:
This seems similar to what `errorHandlingCoordinatorExecutor` already does. Could we reuse it here?

Reply:
I don't see this class, where is it ?

Reply:
https://github.com/apache/flink/blob/5e4efadcae3f89d0661470b754f920a3f0cae337/flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorContext.java#L137

Reply:
thanks. I guess you mean using `ThrowableCatchingRunnable` with the coordinatorExecutor. I totally agree.
Done.